File: backend/app/schemas/claim_intelligence.py
from datetime import datetime
from uuid import UUID

from pydantic import (
    BaseModel,
    ConfigDict,
    Field,
)


class ClaimIntelligenceBase(BaseModel):
    fraud_risk_score: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )

    fraud_risk_level: str | None = None
    fraud_risk_reason: str | None = None

    clinical_review_status: str = "NOT_REVIEWED"
    clinical_review_summary: str | None = None

    sla_status: str = "ON_TRACK"
    sla_due_at: datetime | None = None
    sla_breached: bool = False

    decision_recommendation: str | None = None

    decision_confidence: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )

    decision_reason: str | None = None

    requires_manual_review: bool = True

    reviewed_by: str | None = None
    reviewed_at: datetime | None = None

    model_name: str | None = None
    model_version: str | None = None


class ClaimIntelligenceCreate(
    ClaimIntelligenceBase,
):
    claim_id: UUID


class ClaimIntelligenceUpdate(BaseModel):
    fraud_risk_score: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )

    fraud_risk_level: str | None = None
    fraud_risk_reason: str | None = None

    clinical_review_status: str | None = None
    clinical_review_summary: str | None = None

    sla_status: str | None = None
    sla_due_at: datetime | None = None
    sla_breached: bool | None = None

    decision_recommendation: str | None = None

    decision_confidence: float | None = Field(
        default=None,
        ge=0,
        le=100,
    )

    decision_reason: str | None = None

    requires_manual_review: bool | None = None

    reviewed_by: str | None = None
    reviewed_at: datetime | None = None

    model_name: str | None = None
    model_version: str | None = None


class ClaimIntelligenceResponse(
    ClaimIntelligenceBase,
):
    model_config = ConfigDict(
        from_attributes=True,
    )

    id: UUID
    claim_id: UUID

    created_at: datetime
    updated_at: datetime

File: frontend/src/types/claimIntelligence.ts
export interface ClaimIntelligence {
  id: string;
  claim_id: string;

  fraud_risk_score: number | null;
  fraud_risk_level: string | null;
  fraud_risk_reason: string | null;

  clinical_review_status: string;
  clinical_review_summary: string | null;

  sla_status: string;
  sla_due_at: string | null;
  sla_breached: boolean;

  decision_recommendation: string | null;
  decision_confidence: number | null;
  decision_reason: string | null;

  requires_manual_review: boolean;

  reviewed_by: string | null;
  reviewed_at: string | null;

  model_name: string | null;
  model_version: string | null;

  created_at: string;
  updated_at: string;
}

export interface ClaimIntelligenceCreate {
  claim_id: string;

  fraud_risk_score?: number | null;
  fraud_risk_level?: string | null;
  fraud_risk_reason?: string | null;

  clinical_review_status?: string;
  clinical_review_summary?: string | null;

  sla_status?: string;
  sla_due_at?: string | null;
  sla_breached?: boolean;

  decision_recommendation?: string | null;
  decision_confidence?: number | null;
  decision_reason?: string | null;

  requires_manual_review?: boolean;

  reviewed_by?: string | null;
  reviewed_at?: string | null;

  model_name?: string | null;
  model_version?: string | null;
}

export interface ClaimIntelligenceUpdate {
  fraud_risk_score?: number | null;
  fraud_risk_level?: string | null;
  fraud_risk_reason?: string | null;

  clinical_review_status?: string | null;
  clinical_review_summary?: string | null;

  sla_status?: string | null;
  sla_due_at?: string | null;
  sla_breached?: boolean | null;

  decision_recommendation?: string | null;
  decision_confidence?: number | null;
  decision_reason?: string | null;

  requires_manual_review?: boolean | null;

  reviewed_by?: string | null;
  reviewed_at?: string | null;

  model_name?: string | null;
  model_version?: string | null;
}